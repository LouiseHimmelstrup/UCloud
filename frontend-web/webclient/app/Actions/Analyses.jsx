import { Cloud } from "../../authentication/SDUCloudObject";
import { SET_LOADING, RECEIVE_ANALYSES, SET_PAGE_SIZE } from "../Reducers/Analyses";

export const fetchAnalyses = (analysesPerPage, currentPage) =>
    Cloud.get(`/hpc/jobs/?itemsPerPage=${analysesPerPage}&page=${currentPage}`)
         .then(({response}) => receiveAnalyses(response))

const receiveAnalyses = ({ items, itemsPerPage, pageNumber, pagesInTotal }) => ({
    type: RECEIVE_ANALYSES,
    analyses: items,
    analysesPerPage: itemsPerPage,
    pageNumber: pageNumber,
    totalPages: pagesInTotal
})

export const setPageSize = (pageSize) => ({
    type: SET_PAGE_SIZE,
    pageSize
});

export const setLoading = (loading) => ({
    type: SET_LOADING,
    loading
});