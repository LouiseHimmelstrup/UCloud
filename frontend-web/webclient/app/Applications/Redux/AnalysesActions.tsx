import { Cloud } from "Authentication/SDUCloudObject";
import { SET_ANALYSES_LOADING, RECEIVE_ANALYSES, SET_ANALYSES_ERROR } from "./AnalysesReducer";
import { Page, ReceivePage, SetLoadingAction, Error } from "Types";
import { Analysis } from "..";

/**
 * Fetches a page of analyses based on the itemsPerPage and page provided
 * @param {number} itemsPerPage number of items the retrieved page should contain
 * @param {number} page the page number to be retrieved
 */
export const fetchAnalyses = (itemsPerPage: number, page: number): Promise<ReceivePage<Analysis> | Error> =>
    Cloud.get(`/hpc/jobs/?itemsPerPage=${itemsPerPage}&page=${page}`)
        .then(({ response }) => receiveAnalyses(response)).catch(() =>
            setErrorMessage("Retrieval of analyses failed, please try again later.")
        );

/**
 * Returns an action containing the page retrieved
 * @param {Page<Analysis>} page contains the analyses, pageNumber and items per page
 */
const receiveAnalyses = (page: Page<Analysis>): ReceivePage<Analysis> => ({
    type: RECEIVE_ANALYSES,
    page
});

/**
 * Used to set or remove the error for the component
 * @param {string?} error The error to be renered in the component. Nothing will be rendered if string is null
 * @returns {Error}
 */
export const setErrorMessage = (error?: string): Error => ({
    type: SET_ANALYSES_ERROR,
    error
});

/**
 * Sets whether or not the component is loading
 * @param {boolean} loading - whether or not it is loading
 */
export const setLoading = (loading: boolean): SetLoadingAction => ({
    type: SET_ANALYSES_LOADING,
    loading
});