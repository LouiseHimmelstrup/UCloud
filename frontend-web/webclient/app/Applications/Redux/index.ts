import * as ViewObject from "./ViewObject";
import * as ViewReducer from "./ViewReducer";
import * as ApplicationObject from "./ApplicationsObject";
import * as ApplicationReducer from "./ApplicationsReducer";

export type Reducers = ViewReducer.Reducer & ApplicationReducer.Reducer;
export type Objects = ViewObject.Wrapper & ApplicationObject.Wrapper;

export function init(): Objects {
    return { 
        ...ViewObject.init(),
        ...ApplicationObject.init()
    }
}

export const reducers: Reducers = {
    applicationView: ViewReducer.default,
    applicationsBrowse: ApplicationReducer.default
};