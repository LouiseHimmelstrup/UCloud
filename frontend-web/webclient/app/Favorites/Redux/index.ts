import * as FavoritesObject from "./FavoriteObject";
import * as FavoritesReducer from "./FavoritesReducer";

export type Reducers = FavoritesReducer.Reducer;
export type Objects = FavoritesObject.Wrapper;

export function init(): Objects {
    return {
        ...FavoritesObject.init()
    }
}

export const reducers: Reducers = {
    favorites: FavoritesReducer.default
}