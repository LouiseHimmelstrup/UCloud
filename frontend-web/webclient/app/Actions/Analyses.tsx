import { Cloud } from "../../authentication/SDUCloudObject";
import { failureNotification } from "../UtilityFunctions";
import { SET_ANALYSES_LOADING, RECEIVE_ANALYSES, SET_ANALYSES_PAGE_SIZE } from "../Reducers/Analyses";
import { Page, Analysis } from "../types/types";

export const fetchAnalyses = (analysesPerPage: number, currentPage: number) =>
    Cloud.get(`/hpc/jobs/?itemsPerPage=${analysesPerPage}&page=${currentPage}`)
        .then(({ response }) => receiveAnalyses(response)).catch(() => {
            failureNotification("Retrieval of analyses failed, please try again later.");
            return setLoading(false);
        });

const receiveAnalyses = (page: Page<Analysis>) => ({
    type: RECEIVE_ANALYSES,
    page: page
});

export const setPageSize = (pageSize: number) => ({
    type: SET_ANALYSES_PAGE_SIZE,
    pageSize
});

export const setLoading = (loading: boolean) => ({
    type: SET_ANALYSES_LOADING,
    loading
});