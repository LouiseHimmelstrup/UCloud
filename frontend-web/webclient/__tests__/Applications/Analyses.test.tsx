import * as React from "react";
import Analyses from "Applications/Analyses.tsx";
import { create } from "react-test-renderer";
import { configureStore } from "Utilities/ReduxUtilities";
import { initAnalyses } from "DefaultObjects";
import analyses from "Applications/Redux/AnalysesReducer";
import * as AnalysesActions from "Applications/Redux/AnalysesActions";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router";
import { analyses as analysesPage } from "../mock/Analyses";

describe("Analyses component", () => {
    test("Mount component", () => {
        const store = configureStore({ analyses: initAnalyses() }, { analyses })
        expect(create(
            <Provider store={store}>
                <MemoryRouter>
                    <Analyses />
                </MemoryRouter>
            </Provider>).toJSON()).toMatchSnapshot();
    });

    test.skip("Mount component with non-empty page", () => {
        const store = configureStore({ analyses: initAnalyses() }, { analyses })
        store.getState().analyses.page = analysesPage;
        expect(create(
            <Provider store={store}>
                <MemoryRouter>
                    <Analyses />
                </MemoryRouter>
            </Provider>).toJSON()).toMatchSnapshot();
    });
});