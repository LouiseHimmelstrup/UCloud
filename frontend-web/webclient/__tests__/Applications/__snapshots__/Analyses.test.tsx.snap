// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analyses component Mount component with non-empty page 1`] = `
.c1 {
  margin-left: 190px;
  padding-top: 48px;
  padding-bottom: 14px;
  padding-left: 14px;
  padding-right: 0;
  background-color: #fff;
}

.c4 {
  margin-left: auto;
}

.c11 {
  padding-top: 48px;
}

.c13 {
  cursor: pointer;
}

.c16 {
  margin-bottom: 16px;
  margin-top: 16px;
}

.c19 {
  margin-bottom: 16px;
}

.c20 {
  padding-top: 48px;
  padding-right: 340px;
}

.c8 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  vertical-align: middle;
  cursor: inherit;
  margin-left: .7em;
}

.c28 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  vertical-align: middle;
  cursor: inherit;
}

.c30 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  vertical-align: middle;
  cursor: pointer;
  margin-right: .4em;
}

.c35 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  vertical-align: middle;
  cursor: inherit;
  margin-right: 8px;
  color: #c00;
}

.c36 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  vertical-align: middle;
  cursor: inherit;
  margin-right: 8px;
  color: #00C05A;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  cursor: pointer;
}

.c7 {
  cursor: inherit;
}

.c12 {
  cursor: inherit;
  font-size: 24px;
  margin: 0;
  font-weight: 400;
}

.c14 {
  cursor: inherit;
  font-size: 24px;
  margin: 0;
  margin-top: 16px;
  font-weight: 400;
}

.c15 {
  font-size: 10px;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  display: block;
  margin: 0;
  font-size: 14px;
  color: #000;
  font-weight: 700;
  width: 100%;
}

.c5 {
  position: relative;
  display: inline-block;
}

.c5:hover > div {
  display: block;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 5px;
  border-width: 1px;
  border-style: solid;
  border-color: #c9d3df;
}

.c17 > .c0 {
  width: 100%;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.c17 .sc-fzXfLU {
  border: 0;
  box-shadow: none;
}

.c26 {
  display: inline-block;
  position: relative;
  vertical-align: middle;
  cursor: pointer;
  color: #8393A7;
  margin-right: .5em;
}

.c26 svg[data-name="checked"] {
  display: none;
}

.c26 > input:checked ~ svg[data-name="checked"] {
  display: inline-block;
  color: #006aff;
}

.c26 > input:checked ~ svg[data-name="empty"] {
  display: none;
}

.c27 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  opacity: 0;
  position: absolute;
}

.c2 {
  padding-top: 14px;
  padding-bottom: 14px;
  padding-left: 204px;
  padding-right: 14px;
  width: 100%;
  height: 48px;
  background-color: #fff;
  position: absolute;
  top: 48px;
  left: 0;
  position: fixed;
  z-index: 50;
}

.c10 {
  padding-top: 96px;
  padding-left: 14px;
  padding-right: 14px;
  width: 340px;
  height: 100%;
  position: absolute;
  top: 0;
  right: 0;
  overflow-y: auto;
  position: fixed;
  z-index: 45;
}

.c18 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  width: 100%;
  font-family: inherit;
  color: inherit;
  font-size: 14px;
  background-color: transparent;
  border-radius: 5px;
  border-width: 0px;
  border-style: solid;
  border-color: #c9d3df;
  padding-top: 14px;
  padding-bottom: 14px;
  padding-left: 12px;
  padding-right: 12px;
  margin: 0;
}

.c18::-webkit-input-placeholder {
  color: #8393A7;
}

.c18::-moz-placeholder {
  color: #8393A7;
}

.c18:-ms-input-placeholder {
  color: #8393A7;
}

.c18::placeholder {
  color: #8393A7;
}

.c18::-ms-clear {
  display: none;
}

.c22 {
  background-color: #fff;
  border: 0px;
  border-spacing: 0;
  table-layout: fixed;
  width: 100%;
  min-width: 15em;
  background-color: #fff;
}

.c33 {
  border: 0px;
  border-spacing: 0;
  text-align: center;
}

.c34 {
  border: 0px;
  border-spacing: 0;
}

.c24 {
  cursor: auto;
}

.c24 > .c32 {
  border-spacing: 0;
  border-top: 1px solid rgba(34,36,38,.1);
  padding-top: 8px;
  padding-bottom: 8px;
}

.c31 {
  cursor: pointer;
}

.c31 > .c32 {
  border-spacing: 0;
  border-top: 1px solid rgba(34,36,38,.1);
  padding-top: 8px;
  padding-bottom: 8px;
}

.c23 {
  background-color: #fff;
  padding-top: 11px;
  padding-bottom: 11px;
}

.c25 {
  border-spacing: 0;
  border: 0px;
  text-align: center;
  width: 4%;
  background-color: #fff;
  top: 96px;
  z-index: 10;
  position: -webkit-sticky;
  position: sticky;
}

.c29 {
  border-spacing: 0;
  border: 0px;
  text-align: left;
  background-color: #fff;
  cursor: pointer;
  top: 96px;
  z-index: 10;
  position: -webkit-sticky;
  position: sticky;
}

@media screen and (max-width:511px) {
  .c9 {
    display: none;
  }
}

@media screen and (min-width:512px) and (max-width:639px) {
  .c9 {
    display: none;
  }
}

@media screen and (min-width:640px) and (max-width:767px) {
  .c9 {
    display: none;
  }
}

@media screen and (min-width:768px) and (max-width:1023px) {
  .c21 {
    display: none;
  }
}

@media screen and (min-width:1024px) and (max-width:1279px) {
  .c21 {
    display: none;
  }
}

@media screen and (min-width:1280px) {
  .c21 {
    display: none;
  }
}

<div
  className="c0 c1"
>
  <div
    className="c0 sc-AykKD c2"
    height={48}
    width={1}
  >
    <div
      className="c3"
    >
      <div
        className="c0 c4"
      />
      <div
        className="c5"
        data-tag="dropdown"
      >
        <span
          className="c6"
          cursor="pointer"
          onClick={[Function]}
        >
          <span
            className="c7"
            cursor="inherit"
          >
             
            Jobs per page 25
          </span>
          <svg
            className="c8"
            clipRule="evenodd"
            cursor="inherit"
            fill="currentcolor"
            fillRule="evenodd"
            height=".7em"
            ml=".7em"
            viewBox="0 0 24 21"
            width=".7em"
          >
            <path
              d="M12 20.785L0 0h24L12 20.785z"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
  <div
    className="c0 c9"
  >
    <div
      className="c0 sc-AykKD c10"
      height="100%"
      width={340}
    >
      <div
        className="c0 c11"
      >
        <h3
          className="c12"
          cursor="inherit"
          fontSize={4}
        >
          Quick Filters
        </h3>
        <div
          className="c0 c13"
          cursor="pointer"
          onClick={[Function]}
        >
          <span
            className="c7"
            cursor="inherit"
          >
            Today
          </span>
        </div>
        <div
          className="c0 c13"
          cursor="pointer"
          onClick={[Function]}
        >
          <span
            className="c7"
            cursor="inherit"
          >
            Yesterday
          </span>
        </div>
        <div
          className="c0 c13"
          cursor="pointer"
          onClick={[Function]}
        >
          <span
            className="c7"
            cursor="inherit"
          >
            This week
          </span>
        </div>
        <div
          className="c0 c13"
          cursor="pointer"
          onClick={[Function]}
        >
          <span
            className="c7"
            cursor="inherit"
          >
            No filter
          </span>
        </div>
        <h3
          className="c14"
          cursor="inherit"
          fontSize={4}
        >
          Active Filters
        </h3>
        <label
          className="c15"
          color="black"
          fontSize={1}
          fontWeight="bold"
          width="100%"
        >
          Filter by app state
        </label>
        <div
          className="c5"
          data-tag="dropdown"
        >
          <span
            className="c6"
            cursor="pointer"
            onClick={[Function]}
          >
            Don't filter
            <svg
              className="c8"
              clipRule="evenodd"
              cursor="inherit"
              fill="currentcolor"
              fillRule="evenodd"
              height=".7em"
              ml=".7em"
              viewBox="0 0 24 21"
              width=".7em"
            >
              <path
                d="M12 20.785L0 0h24L12 20.785z"
              />
            </svg>
          </span>
        </div>
        <div
          className="c0 c16"
        >
          <label
            className="c15"
            color="black"
            fontSize={1}
            fontWeight="bold"
            width="100%"
          >
            Job created after
          </label>
          <div
            className="c17"
          >
            <div
              className="react-datepicker-wrapper"
            >
              <div
                className="react-datepicker__input-container"
              >
                <input
                  className="c18"
                  disabled={false}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Don't filter"
                  readOnly={false}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="c0 c19"
        >
          <label
            className="c15"
            color="black"
            fontSize={1}
            fontWeight="bold"
            width="100%"
          >
            Job created before
          </label>
          <div
            className="c17"
          >
            <div
              className="react-datepicker-wrapper"
            >
              <div
                className="react-datepicker__input-container"
              >
                <input
                  className="c18"
                  disabled={false}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Don't filter"
                  readOnly={false}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="c0 c20"
  >
    <div
      className="c0 c21"
    >
      <div
        className="c0 c11"
      >
        <h3
          className="c12"
          cursor="inherit"
          fontSize={4}
        >
          Quick Filters
        </h3>
        <div
          className="c0 c13"
          cursor="pointer"
          onClick={[Function]}
        >
          <span
            className="c7"
            cursor="inherit"
          >
            Today
          </span>
        </div>
        <div
          className="c0 c13"
          cursor="pointer"
          onClick={[Function]}
        >
          <span
            className="c7"
            cursor="inherit"
          >
            Yesterday
          </span>
        </div>
        <div
          className="c0 c13"
          cursor="pointer"
          onClick={[Function]}
        >
          <span
            className="c7"
            cursor="inherit"
          >
            This week
          </span>
        </div>
        <div
          className="c0 c13"
          cursor="pointer"
          onClick={[Function]}
        >
          <span
            className="c7"
            cursor="inherit"
          >
            No filter
          </span>
        </div>
        <h3
          className="c14"
          cursor="inherit"
          fontSize={4}
        >
          Active Filters
        </h3>
        <label
          className="c15"
          color="black"
          fontSize={1}
          fontWeight="bold"
          width="100%"
        >
          Filter by app state
        </label>
        <div
          className="c5"
          data-tag="dropdown"
        >
          <span
            className="c6"
            cursor="pointer"
            onClick={[Function]}
          >
            Don't filter
            <svg
              className="c8"
              clipRule="evenodd"
              cursor="inherit"
              fill="currentcolor"
              fillRule="evenodd"
              height=".7em"
              ml=".7em"
              viewBox="0 0 24 21"
              width=".7em"
            >
              <path
                d="M12 20.785L0 0h24L12 20.785z"
              />
            </svg>
          </span>
        </div>
        <div
          className="c0 c16"
        >
          <label
            className="c15"
            color="black"
            fontSize={1}
            fontWeight="bold"
            width="100%"
          >
            Job created after
          </label>
          <div
            className="c17"
          >
            <div
              className="react-datepicker-wrapper"
            >
              <div
                className="react-datepicker__input-container"
              >
                <input
                  className="c18"
                  disabled={false}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Don't filter"
                  readOnly={false}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="c0 c19"
        >
          <label
            className="c15"
            color="black"
            fontSize={1}
            fontWeight="bold"
            width="100%"
          >
            Job created before
          </label>
          <div
            className="c17"
          >
            <div
              className="react-datepicker-wrapper"
            >
              <div
                className="react-datepicker__input-container"
              >
                <input
                  className="c18"
                  disabled={false}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Don't filter"
                  readOnly={false}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <table
      className="c22"
      width="100%"
    >
      <thead
        className="c23"
      >
        <tr
          className="c24"
          cursor="auto"
        >
          <th
            className="sc-fzXfPE c25"
            width="4%"
          >
            <label
              className="c15"
              color="black"
              fontSize={1}
              fontWeight="bold"
              width="100%"
            >
              <div
                className="c0 c26"
                disabled={false}
              >
                <input
                  checked={false}
                  className="c27"
                  data-tag="masterCheckbox"
                  disabled={false}
                  onChange={[Function]}
                  size={20}
                  type="checkbox"
                />
                <svg
                  className="c28"
                  cursor="inherit"
                  data-name="checked"
                  fill="currentcolor"
                  height={20}
                  viewBox="0 0 24 24"
                  width={20}
                >
                  <path
                    d="M6 3h12c1.7 0 3 1.3 3 3v12c0 1.7-1.3 3-3 3H6c-1.7 0-3-1.3-3-3V6c0-1.7 1.3-3 3-3zm4 14l9-8.6L17.6 7 10 14.3l-3.6-3.5L5 12.2l5 4.8z"
                  />
                </svg>
                <svg
                  className="c28"
                  cursor="inherit"
                  data-name="empty"
                  fill="currentcolor"
                  height={20}
                  viewBox="0 0 24 24"
                  width={20}
                >
                  <path
                    d="M6 5c-.6 0-1 .4-1 1v12c0 .6.4 1 1 1h12c.6 0 1-.4 1-1V6c0-.6-.4-1-1-1H6zm0-2h12c1.7 0 3 1.3 3 3v12c0 1.7-1.3 3-3 3H6c-1.7 0-3-1.3-3-3V6c0-1.7 1.3-3 3-3z"
                  />
                </svg>
              </div>
            </label>
          </th>
          <th
            className="sc-fzXfPE c29"
            onClick={[Function]}
          >
            Name
          </th>
          <th
            className="sc-fzXfPE c29"
            onClick={[Function]}
          >
            State
          </th>
          <th
            className="sc-fzXfPE c29"
            onClick={[Function]}
          >
            Application
          </th>
          <th
            className="sc-fzXfPE c29"
            onClick={[Function]}
          >
            <svg
              className="c30"
              clipRule="evenodd"
              cursor="pointer"
              fill="currentcolor"
              fillRule="evenodd"
              height=".7em"
              mr=".4em"
              viewBox="0 0 18 24"
              width=".7em"
            >
              <path
                d="M9 24l9-10h-6V0H6v14h-6l9 10z"
              />
            </svg>
            Created at
          </th>
          <th
            className="sc-fzXfPE c29"
            onClick={[Function]}
          >
            Expiration
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="c31"
          cursor="pointer"
        >
          <td
            className="c32 c33"
          >
            <div>
              <label
                className="c15"
                color="black"
                fontSize={1}
                fontWeight="bold"
                width="100%"
              >
                <div
                  className="c0 c26"
                  disabled={false}
                >
                  <input
                    checked={false}
                    className="c27"
                    disabled={false}
                    onChange={[Function]}
                    size={20}
                    type="checkbox"
                  />
                  <svg
                    className="c28"
                    cursor="inherit"
                    data-name="checked"
                    fill="currentcolor"
                    height={20}
                    viewBox="0 0 24 24"
                    width={20}
                  >
                    <path
                      d="M6 3h12c1.7 0 3 1.3 3 3v12c0 1.7-1.3 3-3 3H6c-1.7 0-3-1.3-3-3V6c0-1.7 1.3-3 3-3zm4 14l9-8.6L17.6 7 10 14.3l-3.6-3.5L5 12.2l5 4.8z"
                    />
                  </svg>
                  <svg
                    className="c28"
                    cursor="inherit"
                    data-name="empty"
                    fill="currentcolor"
                    height={20}
                    viewBox="0 0 24 24"
                    width={20}
                  >
                    <path
                      d="M6 5c-.6 0-1 .4-1 1v12c0 .6.4 1 1 1h12c.6 0 1-.4 1-1V6c0-.6-.4-1-1-1H6zm0-2h12c1.7 0 3 1.3 3 3v12c0 1.7-1.3 3-3 3H6c-1.7 0-3-1.3-3-3V6c0-1.7 1.3-3 3-3z"
                    />
                  </svg>
                </div>
              </label>
            </div>
          </td>
          <td
            className="c32 c34"
            onClick={[Function]}
          >
            Thomas
          </td>
          <td
            className="c32 c34"
            onClick={[Function]}
          >
            <svg
              className="c35"
              clipRule="evenodd"
              color="#c00"
              cursor="inherit"
              fill="currentcolor"
              fillRule="evenodd"
              height={24}
              isExpired={false}
              mr="8px"
              state="FAILURE"
              viewBox="0 0 25 24"
              width={24}
            >
              <path
                d="M24.002 2.001L22 .001 12 9.6l-10-9.6-2 2 9.6 10L.001 22l2 2 10-9.6 10 9.6 2-2-9.6-10 9.6-9.999z"
                fillRule="nonzero"
              />
            </svg>
            Failure
          </td>
          <td
            className="c32 c34"
            onClick={[Function]}
          >
            
             v
            
          </td>
          <td
            className="c32 c34"
            onClick={[Function]}
          >
            28/08/2018
          </td>
          <td
            className="c32 c34"
            onClick={[Function]}
          >
            N/A
          </td>
        </tr>
        <tr
          className="c31"
          cursor="pointer"
        >
          <td
            className="c32 c33"
          >
            <div>
              <label
                className="c15"
                color="black"
                fontSize={1}
                fontWeight="bold"
                width="100%"
              >
                <div
                  className="c0 c26"
                  disabled={false}
                >
                  <input
                    checked={false}
                    className="c27"
                    disabled={false}
                    onChange={[Function]}
                    size={20}
                    type="checkbox"
                  />
                  <svg
                    className="c28"
                    cursor="inherit"
                    data-name="checked"
                    fill="currentcolor"
                    height={20}
                    viewBox="0 0 24 24"
                    width={20}
                  >
                    <path
                      d="M6 3h12c1.7 0 3 1.3 3 3v12c0 1.7-1.3 3-3 3H6c-1.7 0-3-1.3-3-3V6c0-1.7 1.3-3 3-3zm4 14l9-8.6L17.6 7 10 14.3l-3.6-3.5L5 12.2l5 4.8z"
                    />
                  </svg>
                  <svg
                    className="c28"
                    cursor="inherit"
                    data-name="empty"
                    fill="currentcolor"
                    height={20}
                    viewBox="0 0 24 24"
                    width={20}
                  >
                    <path
                      d="M6 5c-.6 0-1 .4-1 1v12c0 .6.4 1 1 1h12c.6 0 1-.4 1-1V6c0-.6-.4-1-1-1H6zm0-2h12c1.7 0 3 1.3 3 3v12c0 1.7-1.3 3-3 3H6c-1.7 0-3-1.3-3-3V6c0-1.7 1.3-3 3-3z"
                    />
                  </svg>
                </div>
              </label>
            </div>
          </td>
          <td
            className="c32 c34"
            onClick={[Function]}
          >
            Johnny
          </td>
          <td
            className="c32 c34"
            onClick={[Function]}
          >
            <svg
              className="c35"
              clipRule="evenodd"
              color="#c00"
              cursor="inherit"
              fill="currentcolor"
              fillRule="evenodd"
              height={24}
              isExpired={false}
              mr="8px"
              state="FAILURE"
              viewBox="0 0 25 24"
              width={24}
            >
              <path
                d="M24.002 2.001L22 .001 12 9.6l-10-9.6-2 2 9.6 10L.001 22l2 2 10-9.6 10 9.6 2-2-9.6-10 9.6-9.999z"
                fillRule="nonzero"
              />
            </svg>
            Failure
          </td>
          <td
            className="c32 c34"
            onClick={[Function]}
          >
            
             v
            
          </td>
          <td
            className="c32 c34"
            onClick={[Function]}
          >
            29/08/2018
          </td>
          <td
            className="c32 c34"
            onClick={[Function]}
          >
            N/A
          </td>
        </tr>
        <tr
          className="c31"
          cursor="pointer"
        >
          <td
            className="c32 c33"
          >
            <div>
              <label
                className="c15"
                color="black"
                fontSize={1}
                fontWeight="bold"
                width="100%"
              >
                <div
                  className="c0 c26"
                  disabled={false}
                >
                  <input
                    checked={false}
                    className="c27"
                    disabled={false}
                    onChange={[Function]}
                    size={20}
                    type="checkbox"
                  />
                  <svg
                    className="c28"
                    cursor="inherit"
                    data-name="checked"
                    fill="currentcolor"
                    height={20}
                    viewBox="0 0 24 24"
                    width={20}
                  >
                    <path
                      d="M6 3h12c1.7 0 3 1.3 3 3v12c0 1.7-1.3 3-3 3H6c-1.7 0-3-1.3-3-3V6c0-1.7 1.3-3 3-3zm4 14l9-8.6L17.6 7 10 14.3l-3.6-3.5L5 12.2l5 4.8z"
                    />
                  </svg>
                  <svg
                    className="c28"
                    cursor="inherit"
                    data-name="empty"
                    fill="currentcolor"
                    height={20}
                    viewBox="0 0 24 24"
                    width={20}
                  >
                    <path
                      d="M6 5c-.6 0-1 .4-1 1v12c0 .6.4 1 1 1h12c.6 0 1-.4 1-1V6c0-.6-.4-1-1-1H6zm0-2h12c1.7 0 3 1.3 3 3v12c0 1.7-1.3 3-3 3H6c-1.7 0-3-1.3-3-3V6c0-1.7 1.3-3 3-3z"
                    />
                  </svg>
                </div>
              </label>
            </div>
          </td>
          <td
            className="c32 c34"
            onClick={[Function]}
          >
            Fred
          </td>
          <td
            className="c32 c34"
            onClick={[Function]}
          >
            <svg
              className="c35"
              clipRule="evenodd"
              color="#c00"
              cursor="inherit"
              fill="currentcolor"
              fillRule="evenodd"
              height={24}
              isExpired={false}
              mr="8px"
              state="FAILURE"
              viewBox="0 0 25 24"
              width={24}
            >
              <path
                d="M24.002 2.001L22 .001 12 9.6l-10-9.6-2 2 9.6 10L.001 22l2 2 10-9.6 10 9.6 2-2-9.6-10 9.6-9.999z"
                fillRule="nonzero"
              />
            </svg>
            Failure
          </td>
          <td
            className="c32 c34"
            onClick={[Function]}
          >
            
             v
            
          </td>
          <td
            className="c32 c34"
            onClick={[Function]}
          >
            30/08/2018
          </td>
          <td
            className="c32 c34"
            onClick={[Function]}
          >
            N/A
          </td>
        </tr>
        <tr
          className="c31"
          cursor="pointer"
        >
          <td
            className="c32 c33"
          >
            <div>
              <label
                className="c15"
                color="black"
                fontSize={1}
                fontWeight="bold"
                width="100%"
              >
                <div
                  className="c0 c26"
                  disabled={false}
                >
                  <input
                    checked={false}
                    className="c27"
                    disabled={false}
                    onChange={[Function]}
                    size={20}
                    type="checkbox"
                  />
                  <svg
                    className="c28"
                    cursor="inherit"
                    data-name="checked"
                    fill="currentcolor"
                    height={20}
                    viewBox="0 0 24 24"
                    width={20}
                  >
                    <path
                      d="M6 3h12c1.7 0 3 1.3 3 3v12c0 1.7-1.3 3-3 3H6c-1.7 0-3-1.3-3-3V6c0-1.7 1.3-3 3-3zm4 14l9-8.6L17.6 7 10 14.3l-3.6-3.5L5 12.2l5 4.8z"
                    />
                  </svg>
                  <svg
                    className="c28"
                    cursor="inherit"
                    data-name="empty"
                    fill="currentcolor"
                    height={20}
                    viewBox="0 0 24 24"
                    width={20}
                  >
                    <path
                      d="M6 5c-.6 0-1 .4-1 1v12c0 .6.4 1 1 1h12c.6 0 1-.4 1-1V6c0-.6-.4-1-1-1H6zm0-2h12c1.7 0 3 1.3 3 3v12c0 1.7-1.3 3-3 3H6c-1.7 0-3-1.3-3-3V6c0-1.7 1.3-3 3-3z"
                    />
                  </svg>
                </div>
              </label>
            </div>
          </td>
          <td
            className="c32 c34"
            onClick={[Function]}
          >
            Nathan
          </td>
          <td
            className="c32 c34"
            onClick={[Function]}
          >
            <svg
              className="c36"
              clipRule="evenodd"
              color="#00C05A"
              cursor="inherit"
              fill="currentcolor"
              fillRule="evenodd"
              height={24}
              isExpired={false}
              mr="8px"
              state="SUCCESS"
              viewBox="0 0 25 19"
              width={24}
            >
              <path
                d="M8 15L2 8.981l-2 2.02L8 19 24.001 2l-2.002-2L8 15z"
                fillRule="nonzero"
              />
            </svg>
            Success
          </td>
          <td
            className="c32 c34"
            onClick={[Function]}
          >
            
             v
            
          </td>
          <td
            className="c32 c34"
            onClick={[Function]}
          >
            03/09/2018
          </td>
          <td
            className="c32 c34"
            onClick={[Function]}
          >
            N/A
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
